track_local.go

```diff
@@ -49,7 +49,7 @@ type TrackLocal interface {
 	// Bind should implement the way how the media data flows from the Track to the PeerConnection
 	// This will be called internally after signaling is complete and the list of available
 	// codecs has been determined
-	Bind(TrackLocalContext) error
+	Bind(TrackLocalContext) (RTPCodecParameters, error)
```

Comment:
Since the codec selection is done within SDP negotiation, is it possible to have multiple codecs in the context? When there are multiple codecs, does the order represent a priority?

Reply:
This returns the final selection by the `TrackLocal`. This locks in the RTPSender to this selected codec.

You can this in `Send` where `r.codec` is set. Every subsequent call to `Bind` you only get once choice, the codec you selected before.

The first call to `Bind` you can choose any codec you want, you get the entire list. The list will be presented in the order chosen by the remote. I didn't design it that way on purpose though, just was easiest parsing wise.